clippy_utils/src/usage.rs

```diff
@@ -199,3 +199,50 @@ pub fn contains_return_break_continue_macro(expression: &Expr<'_>) -> bool {
     recursive_visitor.visit_expr(expression);
     recursive_visitor.seen_return_break_continue
 }
+
+pub struct UsedAfterExprVisitor<'a, 'tcx> {
+    cx: &'a LateContext<'tcx>,
+    expr: &'tcx Expr<'tcx>,
+    definition: HirId,
+    past_self: bool,
+    used_after_self: bool,
```

Comment:
NIT: I would rename those to 
```suggestion
    past_expr: bool,
    used_after_expr: bool,
```

`self` is confusing to me.